src/mongo_odm/nvp.hpp

```diff
@@ -21,6 +21,24 @@
 #include <tuple>
 #include <type_traits>
 
+#include <bsoncxx/types.hpp>
+
+#include <mongo_odm/util.hpp>
+
+// Preprocessor templates for manipulating multiple arguments.
+#define PP_NARG(...) PP_NARG_(__VA_ARGS__, PP_RSEQ_N())
+#define PP_NARG_(...) PP_ARG_N(__VA_ARGS__)
+#define PP_ARG_N(_1, _2, _3, _4, _5, _6, _7, _8, _9, _10, _11, _12, _13, _14, _15, _16, _17, _18, \
```

Comment:
Ah, it didn't occur to me to take into account the maximum depth of BSON. How about I limit the depth of `MONGO_ODM_CHILD*`, and keep `PP_ARG` as is, in case it gets used later?